### PR TITLE
Add test and lint workflow

### LINKED ISSUE #157: Add Github workflows to build and test
**Is your feature request related to a problem? Please describe.**

When a PR comes through, we have to either manually run builds/tests ourselves or trust that the PR author has run and confirmed they work correctly.

**Describe the solution you'd like**

There should be a Github workflow to run builds and tests automatically for PRs so we can confirm nothing is breaking without putting in additional time validating these manually. We can model something after [Dashboard's workflow](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/.github/workflows/build_and_test_workflow.yml).

**Describe alternatives you've considered**

Keeping the system as-is, but there as it is, time is wasted pulling the changes and running the tests manually.

### PR DESCRIPTION
**Pull request by BSFishy**
### Description
Adds a workflow to lint and test on pushes to branches and on PR's.
 
### Issues Resolved
#157 
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] All tests pass
  - [x] `yarn lint`
  - [x] `yarn test-unit`
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/oui/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: Build and test
+
+on:
+  push:
+    branches: ['**', '!backport/**']
```
Comment (joshuarrrr) [inline .github/workflows/build_and_test.yml:5]: also running unit tests on backports may provide some additional confidence against regressions, but I think you can also make a good case to exclude them.

**Review by joshuarrrr — DISMISSED** [review]

File: .github/workflows/build_and_test.yml (hunk 1 above)
Comment (AMoo-Miki) [reply .github/workflows/build_and_test.yml:5]: I am with Josh on having tests run on backports.

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    paths-ignore:
+      - '**/*.md'
+  pull_request:
+    branches: ['**']
+    paths-ignore:
+      - '**/*.md'
+
+jobs:
+  build-lint-test:
+    runs-on: ubuntu-latest
```
Comment (AMoo-Miki) [inline .github/workflows/build_and_test.yml:15]: Do we know that OUI builds on Windows? Should we add a `build and Verify` for Windows too?

Comment (BSFishy) [reply .github/workflows/build_and_test.yml]: Added Windows workflow :+1:

**Review by AMoo-Miki — APPROVED** [review]

**Review by AMoo-Miki — APPROVED** [review]

File: .github/workflows/build_and_test.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        with:
+          node-version-file: '.nvmrc'
+          registry-url: 'https://registry.npmjs.org'
+          cache: 'yarn'
+
+      - name: Setup Yarn
+        run: |
+          npm uninstall -g yarn
+          npm i -g yarn@1.22.10
+          yarn install --frozen-lockfile
```
Comment (ashwin-pc) [inline .github/workflows/build_and_test.yml:35]: This seems unnecessary, see https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#caching-packages-data. The previous action should already install and cache yarn correctly. I havent validated this, but this seems to be correct based on the documentation for `setup-node`

Comment (BSFishy) [reply .github/workflows/build_and_test.yml:35]: Reading through the docs and a little bit of the source code for the action, it looks like it essentially is just a shorthand to use the [`actions/cache`](https://github.com/actions/cache/blob/main/examples.md#node---npm) action on `node_modules` and `yarn.lock`, meaning that if we update dependencies or change dependencies, those changes wouldn't be reflected in the workflow.

**Review by ashwin-pc — APPROVED** [review]

**Pull request merged**